client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -55,31 +67,50 @@ export class SimpleSiteRenameForm extends Component {
 		this.props.requestSiteRename( selectedSiteId, this.state.domainFieldValue, discard );
 	};
 
-	getDomainValidationMessage( domain ) {
+	setValidationState = () => {
 		const { translate } = this.props;
+		const { domainFieldValue } = this.state;
 
-		if ( isEmpty( domain ) ) {
-			return '';
+		let validationProperties = {
+			showValidationMessage: false,
+			validationMessage: '',
+		};
+
+		if ( isEmpty( domainFieldValue ) ) {
+			this.setState( validationProperties );
+			return;
 		}
 
-		if ( domain.match( /[^a-z0-9]/i ) ) {
-			return translate( 'Site address can only contain letters (a-z) and numbers.' );
+		if ( domainFieldValue.match( /[^a-z0-9]/i ) ) {
+			validationProperties = {
+				showValidationMessage: true,
+				validationMessage: translate( 'Site address can only contain letters (a-z) and numbers.' ),
+			};
 		}
 
-		if ( ! inRange( domain.length, SUBDOMAIN_LENGTH_MINIMUM, SUBDOMAIN_LENGTH_MAXIMUM ) ) {
-			return translate(
-				'Site address length should be between %(minimumLength)s and %(maximumLength)s characters.',
-				{
-					args: {
-						minimumLength: SUBDOMAIN_LENGTH_MINIMUM,
-						maximumLength: SUBDOMAIN_LENGTH_MAXIMUM,
-					},
-				}
-			);
+		if (
+			! inRange( domainFieldValue.length, SUBDOMAIN_LENGTH_MINIMUM, SUBDOMAIN_LENGTH_MAXIMUM )
+		) {
+			validationProperties = {
+				showValidationMessage: domainFieldValue.length > SUBDOMAIN_LENGTH_MAXIMUM,
+				validationMessage: translate(
+					'Site address length should be between %(minimumLength)s and %(maximumLength)s characters.',
+					{
+						args: {
+							minimumLength: SUBDOMAIN_LENGTH_MINIMUM,
+							maximumLength: SUBDOMAIN_LENGTH_MAXIMUM,
+						},
+					}
+				),
+			};
 		}
 
-		return '';
-	}
+		this.setState( validationProperties, () => {
+			this.state.validationMessage
+				? this.debouncedShowValidationMessage()
+				: this.debouncedValidationCheck();
+		} );
+	};
 
 	showConfirmationDialog() {
```

Comment:
Am I right in thinking `showConfirmationDialog` is no longer used?

Reply:
That was a hiccup of mine. I'd skipped that step in testing and forgotten to swap it back 😬 That's taken care of now :)